src/main/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotChangeEventSource.java

```diff
@@ -494,21 +494,11 @@ protected SnapshotResult<YugabyteDBOffsetContext> doExecute(ChangeEventSourceCon
                                           resp.getWriteId(), resp.getSnapshotTime());
                 LOGGER.debug("Final OpId for tablet {} is {}", part.getId(), finalOpId);
 
-                // In cases where the tablet is empty, the response checkpoint can still move ahead and we should
-                // also move the explicit checkpoint forward, given that it was already greater than the lsn of the last seen valid record.
-                // Otherwise, we will be stuck waiting for callback on an empty tablet.
-                if (taskContext.shouldEnableExplicitCheckpointing()) {
-                  // If the response doesn't have any record and we got the snapshot end marker, we know the snapshot is empty.
-                  SourceInfo sourceInfo = previousOffset.getSourceInfo(part);
-                  if (isSnapshotCompleteMarker(finalOpId) && sourceInfo.noRecordSeen()) {
-                    LOGGER.info("Should not wait for callback on tablet {}", part.getId());
-                    shouldWaitForCallback.put(part.getId(), Boolean.FALSE);
-                  }
-
-                  OpId lastRecordCheckpoint = sourceInfo.lastRecordCheckpoint();
-                  if (sourceInfo.noRecordSeen() || lastRecordCheckpoint.isLesserThanOrEqualTo(explicitCdcSdkCheckpoint)) {
-                    tabletToExplicitCheckpoint.put(part.getId(), finalOpId.toCdcSdkCheckpoint());
-                  }
+                // If the response doesn't have any record, it is safe to assume that we should not wait
+                // for the callback to come and that we can proceed further in processing this particular tablet.
+                if (previousOffset.getSourceInfo(part).noRecordSeen()) {
+                  LOGGER.info("Should not wait for callback on tablet {}", part.getId());
+                  shouldWaitForCallback.put(part.getId(), Boolean.FALSE);
```

Comment:
shouldWaitForCallback is map with Boolean value? Shouldn't we use Set for it?

Reply:
We can use. Previously it was used in other contexts as well, I will change this to use set.